features/FEATURE_BLE/source/gap/AdvertisingDataBuilder.cpp

```diff
@@ -0,0 +1,469 @@
+    adv_data_type_t advDataType,
+    mbed::Span<const uint8_t> fieldData
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return replaceField(advDataType, fieldData, field);
+    } else {
+        return BLE_ERROR_NOT_FOUND;
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::appendData(
+    adv_data_type_t advDataType,
+    mbed::Span<const uint8_t> fieldData
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return appendToField(fieldData, field);
+    } else {
+        return BLE_ERROR_NOT_FOUND;
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::removeData(
+    adv_data_type_t advDataType
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return removeField(field);
+    } else {
+        return BLE_ERROR_NOT_FOUND;
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::addOrReplaceData(
+    adv_data_type_t advDataType,
+    mbed::Span<const uint8_t> fieldData
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return replaceField(advDataType, fieldData, field);
+    } else {
+        return addField(advDataType, fieldData);
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::addOrAppendData(
+    adv_data_type_t advDataType,
+    mbed::Span<const uint8_t> fieldData
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return appendToField(fieldData, field);
+    } else {
+        return addField(advDataType, fieldData);
+    }
+}
+
+void AdvertisingDataBuilder::clear() {
+    memset(_buffer.data(), 0, _buffer.size());
+    _payload_length = 0;
+}
+
+ble_error_t AdvertisingDataBuilder::setAppearance(
+    adv_data_appearance_t appearance
+) {
+    uint8_t appearence_byte = appearance.value();
+    mbed::Span<const uint8_t> appearance_span((const uint8_t*) &appearence_byte, 2);
+    return addOrReplaceData(adv_data_type_t::APPEARANCE, appearance_span);
+}
+
+ble_error_t AdvertisingDataBuilder::setFlags(
+    adv_data_flags_t flags
+) {
+    uint8_t flags_byte = flags.value();
+    mbed::Span<const uint8_t> flags_span((const uint8_t*) &flags_byte, 1);
+    return addOrReplaceData(adv_data_type_t::FLAGS, flags_span);
+}
+
+ble_error_t AdvertisingDataBuilder::setTxPowerAdvertised(
+    advertising_power_t txPower
+) {
+    mbed::Span<const uint8_t> power_span((const uint8_t*) &txPower, 1);
+    return addOrReplaceData(adv_data_type_t::TX_POWER_LEVEL, power_span);
+}
+
+ble_error_t AdvertisingDataBuilder::setName(
+    const char* name,
+    bool complete
+) {
+    mbed::Span<const uint8_t> power_span((const uint8_t*)name, strlen(name));
+
+    if (complete) {
+        return addOrReplaceData(adv_data_type_t::COMPLETE_LOCAL_NAME, power_span);
+    } else {
+        return addOrReplaceData(adv_data_type_t::SHORTENED_LOCAL_NAME, power_span);
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::setManufacturerSpecificData(
+    mbed::Span<const uint8_t> data
+) {
+    if (data.size() < 2) {
```

Comment:
Why < 2 ? What does this mean ? Should be commented or #defined

Reply:
Fixed.